Sources/WebAuthn/Ceremonies/Registration/PublicKeyCredentialCreationOptions.swift

```diff
@@ -55,17 +77,87 @@ public struct PublicKeyCredentialCreationOptions: Encodable, Sendable {
         try container.encode(user, forKey: .user)
         try container.encode(relyingParty, forKey: .relyingParty)
         try container.encode(publicKeyCredentialParameters, forKey: .publicKeyCredentialParameters)
-        try container.encodeIfPresent(timeout?.milliseconds, forKey: .timeout)
+        try container.encodeIfPresent(timeoutInMilliseconds, forKey: .timeoutInMilliseconds)
         try container.encode(attestation, forKey: .attestation)
+        try container.encode(authenticatorSelection, forKey: .authenticatorSelection)
+        try container.encode(hints, forKey: .hints)
+        try container.encode(extensions, forKey: .extensions)
+        try container.encode(excludeCredentials, forKey: .excludeCredentials)
+    }
+    
+    let hints: [Hint]
+    enum Hint: String, Encodable
+    {
+        /// Hint to get the user to register their platform authenticator
+        case clientDevice = "client-device"
+        
+        /// Hint to the user should be guided to register a security key. Iconography and text should emphasize the use of security keys
+        case securityKey = "security-key"
+        
+        /// Hint to the user to to register a passkey using their mobile device by scanning a QR code that’s displayed on a computer
+        case hybrid = "hybrid"
+    }
+    
+    let extensions: Extensions
+    
+    struct Extensions: Encodable
+    {
+        let credProps: Bool
```

Comment:
Could we use a CodingKeys declaration to give this type a nicer name in swift? Without looking into the spec, I have no clue what this refers to.

Reply:
Yeah that is a good idea to use codingkeys to have a better name in the library for this. Currently I absolutely have not enough understanding of the spec to be able to convert the below statement into something more descriptive. Any suggestions?

> WebAuthn Extension Identifiers defines the initial set of extensions to be registered in the [IANA Registry](https://ldapwiki.com/wiki/Wiki.jsp?page=IANA%20Registry) "WebAuthn Extension Identifier" registry established by [WebAuthn-Registries](https://ldapwiki.com/wiki/Wiki.jsp?page=WebAuthn-Registries).

These [MAY](https://ldapwiki.com/wiki/Wiki.jsp?page=MAY) be implemented by [User-agents](https://ldapwiki.com/wiki/Wiki.jsp?page=User-agent) targeting broad interoperability.

Reply:
https://www.w3.org/TR/webauthn-2/#sctn-authenticator-credential-properties-extension

I would suggest `credentialPropertiesExtension`, set to a `CredentialPropertiesExtension.RequestOptions` type with `.requested` and `.ignore` options that render the appropriate value to the encoded output (either `true`, or the key is skipped)

Reply:
Not sure I follow. Conceptually, do you mean something like the following (except instead of enum I would use `UnreferencedStringEnumeration`):

```
let credentialPropertiesExtension: CredentialPropertiesExtension
struct CredentialPropertiesExtension: Encodable
{
    let options: RequestOptions
    
    enum RequestOptions: Encodable
    {
        case requested
        case ignored
    }
}
```